pilot/pkg/model/push_context.go

```diff
@@ -1903,16 +1905,75 @@ func (ps *PushContext) initEnvoyFilters(env *Environment) {
 		return in < jn
 	})
 
-	ps.envoyFiltersByNamespace = make(map[string][]*EnvoyFilterWrapper)
 	for _, envoyFilterConfig := range envoyFilterConfigs {
 		efw := convertToEnvoyFilterWrapper(&envoyFilterConfig)
-		if _, exists := ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace]; !exists {
-			ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace] = make([]*EnvoyFilterWrapper, 0)
-		}
 		ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace] = append(ps.envoyFiltersByNamespace[envoyFilterConfig.Namespace], efw)
 	}
 }
 
+// pre computes envoy filters per namespace
+func (ps *PushContext) updateEnvoyFilters(env *Environment, changedEnvoyFilters []ConfigKey, previousIndex map[string][]*EnvoyFilterWrapper) {
+	// copy old index
+	ps.envoyFiltersByNamespace = make(map[string][]*EnvoyFilterWrapper, len(previousIndex))
+	for k, v := range previousIndex {
+		ps.envoyFiltersByNamespace[k] = v
```

Comment:
```suggestion
		ps.envoyFiltersByNamespace[k] = slices.Clone(v)
```

Seems preferable to do it here instead of L1928; the latter is in a loop?

I suppose you avoid copy for namespaces we don't modify, although its just a copy of the pointers so not terrible. We could precompute the set of namespaces changed and only clone them if we really want.

if not, these 4 lines can just be `ps.envoyFiltersByNamespace = maps.Clone(previousIndex)` btw

Reply:
yes, i was intentionally doing it in the loop to avoid copying slices we wont alter